applications/dashboard/models/class.userauthenticationnoncemodel.php

```diff
@@ -1,25 +1,34 @@
 <?php
 /**
- * UserAuthenticationNonce model.
- *
+ * @author Chris Chabilall chris.c@vanillaforums.com
  * @copyright 2009-2018 Vanilla Forums Inc.
  * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
  */
 
 /**
- * Handles user data.
+ * UserAuthenticationNonce model.
+ * Handles user data and issuing and consuming nonce's.
  */
 class UserAuthenticationNonceModel extends Gdn_Model {
 
-    use \Vanilla\PrunableTrait;
+    use \Vanilla\PrunableTrait, \Vanilla\TokenSigningTrait;
+
+    /**
+     * @var string $tokenIdentifier Used to deteremine what type of token is generated.
+     */
+    protected static $tokenIdentifier = "nonce";
 
     /**
      * Class constructor. Defines the related database table name.
+     *
+     * @param string $secret The secret used to sign access tokens for the client.
      */
-    public function __construct() {
+    public function __construct($secret = '') {
         parent::__construct('UserAuthenticationNonce');
         $this->setPruneField('Timestamp');
         $this->setPruneAfter('45 minutes');
+        $this->secret = $secret ?: c('Garden.Cookie.Salt');
```

Comment:
Use the trait's setter for this property. Don't directly access it (it should be private).